packages/platform-core/src/usage-data/cli_telemetry_data.ts

```diff
@@ -0,0 +1,70 @@
+// Core Identifiers
+export type CoreIdentifiersDetails = {
+  sessionUuid: string;
+  timestamp: string;
+  localProjectId: string;
+  accountId?: string;
+  payloadVersion: string;
+  awsRegion: string;
+};
+
+// Command Details
+export type CommandDetails = {
+  success: boolean;
+  command: {
+    path: string[];
+    parameters: string[];
+  };
+  executionTime: number;
+};
+
+// Environment Details
+export type EnvironmentDetails = {
+  osRelease: string;
+  shell: string;
+  npmUserAgent: string;
+  ci: boolean;
+  memory: {
+    total: number;
+    free: number;
+  };
+};
+
+// Latency (Optional)
+export type Latency = {
+  init?: number;
+  synthesis?: number;
+  deployment?: number;
+  hotSwap?: number;
+};
+
+// Error Details (Optional)
+export type ErrorDetails = {
+  primary: {
+    name: string;
+    message: string;
+  };
+  cause?: Array<{
+    name: string;
+    message: string;
+    methodName: string;
+    file: string;
+    lineNumber: number;
+    columnNumber: string;
+  }>;
+};
```

Comment:
```suggestion
// Error Details (Optional)
export type ErrorDetails = {
  name: string;
  message: string;
  stack: string;
  cause?: ErrorDetails;
};
```

@Amplifiyer wdyt?

Reply:
yup, that's what I had in mind. We need to validate this with a POC though

Reply:
Updated, it !

Will look something like this
export type ErrorDetails = {
  name: string;
  message: string;
  stack: string;
  cause?: ErrorDetails;
};

const errorDetails: ErrorDetails = {
  name: "Error",
  message: "API layer error",
  stack: `Error: API layer error
    at throwLevel3 (/Users/vigy/Documents/GitHub/amplify-backend/packages/cli-core/src/printer/printer.test.ts:113:15)
    at TestContext.<anonymous> (/Users/vigy/Documents/GitHub/amplify-backend/packages/cli-core/src/printer/printer.test.ts:129:7)
    at async Test.run (node:internal/test_runner/test:632:9)
    at async Suite.processPendingSubtests (node:internal/test_runner/test:374:7)`,
  cause: {
    name: "Error",
    message: "Service layer error",
    stack: `Error: Service layer error
      at throwLevel2 (/Users/vigy/Documents/GitHub/amplify-backend/packages/cli-core/src/printer/printer.test.ts:106:15)
      at throwLevel3 (/Users/vigy/Documents/GitHub/amplify-backend/packages/cli-core/src/printer/printer.test.ts:111:9)
      at TestContext.<anonymous> (/Users/vigy/Documents/GitHub/amplify-backend/packages/cli-core/src/printer/printer.test.ts:129:7)
      at async Test.run (node:internal/test_runner/test:632:9)
      at async Suite.processPendingSubtests (node:internal/test_runner/test:374:7)`,
    cause: {
      name: "Error",
      message: "Data layer error",
      stack: `Error: Data layer error
        at throwLevel1 (/Users/vigy/Documents/GitHub/amplify-backend/packages/cli-core/src/printer/printer.test.ts:99:15)
        at throwLevel2 (/Users/vigy/Documents/GitHub/amplify-backend/packages/cli-core/src/printer/printer.test.ts:104:9)
        at throwLevel3 (/Users/vigy/Documents/GitHub/amplify-backend/packages/cli-core/src/printer/printer.test.ts:111:9)
        at TestContext.<anonymous> (/Users/vigy/Documents/GitHub/amplify-backend/packages/cli-core/src/printer/printer.test.ts:129:7)
        at async Test.run (node:internal/test_runner/test:632:9)
        at async Suite.processPendingSubtests (node:internal/test_runner/test:374:7)`,
      cause: {
        name: "Error",
        message: "Database connection failed",
        stack: `Error: Database connection failed
          at throwLevel1 (/Users/vigy/Documents/GitHub/amplify-backend/packages/cli-core/src/printer/printer.test.ts:97:15)
          at throwLevel2 (/Users/vigy/Documents/GitHub/amplify-backend/packages/cli-core/src/printer/printer.test.ts:104:9)
          at throwLevel3 (/Users/vigy/Documents/GitHub/amplify-backend/packages/cli-core/src/printer/printer.test.ts:111:9)
          at TestContext.<anonymous> (/Users/vigy/Documents/GitHub/amplify-backend/packages/cli-core/src/printer/printer.test.ts:129:7)
          at async Test.run (node:internal/test_runner/test:632:9)
          at async Suite.processPendingSubtests (node:internal/test_runner/test:374:7)`
      }
    }
  }
};